pkg/apis/core/validation/shoot_test.go

```diff
@@ -1661,6 +1661,100 @@ var _ = Describe("Shoot Validation Tests", func() {
 			// uniqueness by key/effect
 			Entry("not unique", []corev1.Taint{{Key: "foo", Value: "bar", Effect: corev1.TaintEffectNoSchedule}, {Key: "foo", Value: "baz", Effect: corev1.TaintEffectNoSchedule}}, field.ErrorTypeDuplicate),
 		)
+
+		It("should reject if data volume size does not match size regex", func() {
+			maxSurge := intstr.FromInt(1)
+			maxUnavailable := intstr.FromInt(0)
+			name1 := "vol1-name"
+			name2 := "vol2-name"
+			dataVolumes := []core.Volume{{Name: &name1, Size: "75Gi"}, {Name: &name2, Size: "12MiB"}}
+			worker := core.Worker{
+				Name: "worker-name",
+				Machine: core.Machine{
+					Type: "large",
+				},
+				MaxSurge:       &maxSurge,
+				MaxUnavailable: &maxUnavailable,
+				DataVolumes:    dataVolumes,
+			}
+			errList := ValidateWorker(worker, nil)
+			Expect(errList).To(HaveLen(1))
```

Comment:
Instead of using `HaveLen()` and checking indices of `errList`, can you use `ConsistOf` like in the newer other tests?